internal/service/console.go

```diff
@@ -0,0 +1,40 @@
+package service
+
+import (
+	"context"
+
+	"github.com/flightctl/flightctl/internal/api/server"
+	"github.com/flightctl/flightctl/internal/flterrors"
+	"github.com/flightctl/flightctl/internal/store"
+	"github.com/google/uuid"
+)
+
+func (h *ServiceHandler) RequestConsole(ctx context.Context, request server.RequestConsoleRequestObject) (server.RequestConsoleResponseObject, error) {
+	orgId := store.NullOrgId
+
+	// make sure the device exists
+	_, err := h.store.Device().Get(ctx, orgId, request.Name)
+	if err != nil {
+		switch err {
+		case flterrors.ErrResourceNotFound:
+			return server.RequestConsole404JSONResponse{}, nil
+		default:
+			return nil, err
+		}
+	}
+
+	sessionId := uuid.New().String()
+
+	annotations := map[string]string{"flightctl.io/console": sessionId}
+
+	if err := h.store.Device().UpdateAnnotations(ctx, orgId, request.Name, annotations, []string{}); err != nil {
+		return server.RequestConsole401JSONResponse{Message: "Unable to annotate device for console setup"}, err
```

Comment:
Why a 401 error?

Reply:
hmm, yes, may be an error 500 would be the right thing, this is not about authorization.